src/connector/src/sink/remote.rs

```diff
@@ -0,0 +1,574 @@
+        DataType::Float64 => TypeName::Float,
+        DataType::Time => TypeName::Time,
+        DataType::Date => TypeName::Date,
+        DataType::Decimal => TypeName::Decimal,
+        DataType::Timestamp => TypeName::Timestamp,
+        DataType::Timestampz => TypeName::Timestampz,
+        DataType::Interval => TypeName::Interval,
+        DataType::List { datatype: _ } => TypeName::List,
+        DataType::Struct(_) => TypeName::Struct,
+        DataType::Varchar => TypeName::Varchar,
+    }
+}
+
+#[async_trait]
+impl Sink for RemoteSink {
+    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
+        let mut row_ops = vec![];
+        for (op, row_ref) in chunk.rows() {
+            let mut map = serde_json::Map::new();
+            row_ref
+                .values()
+                .zip_eq(self.schema.fields.iter())
+                .for_each(|(v, f)| {
+                    map.insert(f.name.clone(), parse_datum(v));
+                });
+            let row_op = RowOp {
+                op_type: parse_stream_op(op),
+                line: serde_json::to_string(&map).unwrap(),
+            };
+
+            row_ops.push(row_op);
+        }
+
+        let epoch = self.epoch.ok_or_else(|| {
+            SinkError::Remote("epoch has not been initialize, call `begin_epoch`".to_string())
+        })?;
+        let batch_id = self.batch_id;
+        self.on_sender_alive()?
+            .send(SinkTask {
+                request: Some(SinkRequest::Write(WriteBatch {
+                    epoch,
+                    batch_id,
+                    payload: Some(Payload::JsonPayload(JsonPayload { row_ops })),
+                })),
+            })
+            .map_err(|e| SinkError::Remote(e.to_string()))?;
+        self.response_stream
+            .on_response_ok()
+            .await
+            .map(|_| self.batch_id += 1)
+    }
+
+    async fn begin_epoch(&mut self, epoch: u64) -> Result<()> {
+        self.on_sender_alive()?
+            .send(SinkTask {
+                request: Some(SinkRequest::StartEpoch(StartEpoch { epoch })),
+            })
+            .map_err(|e| SinkError::Remote(e.to_string()))?;
+        self.response_stream
+            .on_response_ok()
+            .await
+            .map(|_| self.epoch = Some(epoch))
+    }
+
+    async fn commit(&mut self) -> Result<()> {
+        let epoch = self.epoch.ok_or_else(|| {
+            SinkError::Remote("epoch has not been initialize, call `begin_epoch`".to_string())
+        })?;
+        self.on_sender_alive()?
+            .send(SinkTask {
+                request: Some(SinkRequest::Sync(SyncBatch { epoch })),
+            })
+            .map_err(|e| SinkError::Remote(e.to_string()))?;
+        self.response_stream.on_response_ok().await.map(|_| ())
+    }
+
+    async fn abort(&mut self) -> Result<()> {
+        self.request_sender = None;
+        Ok(())
+    }
+}
+
+fn parse_stream_op(op: stream_chunk::Op) -> i32 {
```

Comment:
The `stream_chunk::Op` has a method called `to_protobuf`, which is the same the this function.